### DISCUSSION TITLE
Infinite scrolling with mutation

### DISCUSSION THREAD

**Discussion by iammemon** [urql-graphql/urql#2864] [Q&A]
**Problem statement:** I have a use case where I need to implement infinite scrolling with cursor pagination with additional requirements
-  item in the list can be deleted  
- data polling after every x second 

I did some experiments with [this ](https://formidable.com/open-source/urql/docs/basics/ui-patterns/#infinite-scrolling) approach and was able to make it work for my use case but also notice the following cons
- didn't like the fact every time the UI needs to be changed to implement infinite scrolling
- doesn't work well when you have a layout component that displays different states based on props like fetching or error and doesn't render child components(in this case pages) when those states are actives 
- code becomes awkward to handle mutations

we use document cache in our app so haven't explored the relayPagination exchange option but I haven't seen any docs for how it handles mutation or stale data

I am looking for a solution similar to what react-query is [offering](https://tanstack.com/query/v4/docs/guides/infinite-queries#what-happens-when-an-infinite-query-needs-to-be-refetched)

what is the recommended way to do it in this use case?

**Comment by JasperAlexander** [comment]
I'm having the same problem. Do you already have a solution?
